plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java

```diff
@@ -410,6 +410,21 @@ public void testDropAndAddColumnWithSameName()
                 .hasMessageContaining("Cannot drop column from table using column mapping mode NONE");
     }
 
+    @Test(dataProvider = "columnMappingModeDataProvider")
+    public void testDropPartitionColumn(ColumnMappingMode mode)
```

Comment:
One small ask:
In case of the product tests which are being replaced with integration tests, please do this in a separate commit to distinguish easier what the changes in TestDeltaLakeColumnMappingMode.